File: data structures and algorithms/java/08_031.java
import java.util.HashSet;

public class RangeDuplicateFinder {
    public static boolean hasNearbyDuplicates(int[] nums, int k) {
        HashSet<Integer> window = new HashSet<>();
        
        for (int i = 0; i < nums.length; i++) {
            if (i >= k) {
                window.remove(nums[i - k]);
            }
            
            if (window.contains(nums[i])) {
                return true;
            }
            
            window.add(nums[i]);
        }
        
        return false;
    }
}

File: data structures and algorithms/cs/08_031.cs
using System.Collections.Generic;

public class RangeDuplicateFinder {
    public static bool HasNearbyDuplicates(int[] nums, int k) {
        var window = new HashSet<int>();

        for (int i = 0; i < nums.Length; i++) {
            if (i >= k) {
                window.Remove(nums[i - k]);
            }

            if (window.Contains(nums[i])) {
                return true;
            }

            window.Add(nums[i]);
        }

        return false;
    }
}